### PR TITLE
paredit: added Minor Emacs mode for performing structured editing of …

### PR DESCRIPTION
**Pull request by ghost**
…S-expression data

Hi, could somebody please merge this?  Thank you!

### CODE REVIEW

**Comment by pSub** [comment]
Please rename the commit message according to CONTRIBUTING.md.

File: pkgs/applications/editors/emacs-modes/paredit/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{stdenv, fetchgit, emacs}:
+
+stdenv.mkDerivation {
+  name = "paredit";
```
Comment (pSub) [inline pkgs/applications/editors/emacs-modes/paredit/default.nix]: Please append a version (or if there are no versions a date).

File: pkgs/applications/editors/emacs-modes/paredit/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  buildInputs = [emacs];
+  buildPhase = ''
+    emacs --batch --eval '(byte-compile-file "paredit.el" t)'
+  '';
+  installPhase = ''
+    mkdir -p $out/share/emacs/site-lisp
+    cp paredit.elc  $out/share/emacs/site-lisp
+  '';
+  meta = {
+    homepage = "http://www.emacswiki.org/emacs/ParEdit";
```
Comment (pSub) [inline pkgs/applications/editors/emacs-modes/paredit/default.nix]: You don't need quotes for URLs.

**Comment by oxij** [comment]
If you don't mind I'd ask you to implement this for emacsPackagesNg in
emacs-packages.nix. It would take half the lines of code.

**Comment by tohl** [comment]
OK, I have implemented suggestions by Pascal but I'll try
emacsPackagesNg then, when I find time.

**Comment by oxij** [comment]
All you need is to add

  paredit = trivialBuild rec {
    pname = "paredit-${version}";
    version = "25";
    src = fetchgit {
      url = http://mumble.net/~campbell/git/paredit.git/;
      rev = "9a696fdcce87c9d9eec4569a9929d0300ac6ae5c";
      sha256 = "13wjqimp2s6pwcqix8pmsrk76bq1cxlnwmj3m57bb5y60y67vp9l";
    };
    meta = {
      homepage = http://www.emacswiki.org/emacs/ParEdit;
      description = "Minor Emacs mode for performing structured editing of S-expression data";
      license = gpl3Plus;
    };
  };

to top-level/emacs-packages.nix and should be done.

**Comment by tohl** [comment]
I have updated the pull request as you suggested.  Thanks Jan!

**Comment by oxij** [comment]
Yay, thank you!

:+1:

**Pull request merged**